src/library_syscall.js

```diff
@@ -923,11 +913,13 @@ var SyscallsLibrary = {
     err('warning: untested syscall');
 #endif
     path = SYSCALLS.getStr(path);
+    var nofollow = flags & {{{ cDefine('AT_SYMLINK_NOFOLLOW') }}};
+    flags = flags & (~{{{ cDefine('AT_SYMLINK_NOFOLLOW') }}});
 #if ASSERTIONS
     assert(flags === 0);
 #endif
     path = SYSCALLS.calculateAt(dirfd, path);
-    FS.chown(path, owner, group);
+    (nofollow ? FS.lchown : FS.chown)(path, owner, group);
```

Comment:
I wonder if its worth refactoring the FS api such that nofollow is a param there too?

Reply:
I think that is done internally:

https://github.com/emscripten-core/emscripten/blob/b2b97c22217e8b2e795f0aa5f82f3ca2ec529f74/src/library_fs.js#L927-L929

Perhaps the idea was to have the JS API mirror the POSIX API as much as possible, then internally it becomes a boolean. I am not sure though - that was many years ago and I didn't write the code. But I can see a good case both ways.

In any case as this is for the old FS, maybe best to leave it?

Reply:
Sure, sounds good.